beacon/pow/src/main/java/tech/pegasys/teku/beacon/pow/DepositSnapshotFileLoader.java

```diff
@@ -63,10 +83,95 @@ private Optional<DepositTreeSnapshot> loadDepositSnapshot(
         });
   }
 
-  private DepositTreeSnapshot loadFromResource(final String source) throws IOException {
-    return DepositTreeSnapshot.fromBytes(
-        ResourceLoader.urlOrFile("application/octet-stream")
-            .loadBytes(source)
-            .orElseThrow(() -> new FileNotFoundException("Not found")));
+  private DepositTreeSnapshot loadFromUrl(final String path) throws IOException {
```

Comment:
We've wound up making this process significantly more complex than I think it really needs to be.

Firstly we can just use `ResourceLoader.urlOrFile` to load the resource  as we do in `ChainDataLoader.loadState`. The default values pointing to the class path would be specified as a URL as like in `Eth2NetworkConfiguration.Builder.genesisStateFromClasspath`.

Then there's a lot of complexity added by supporting both JSON and SSZ formats, particularly when dealing with the extra data wrapper the REST API adds. I can see the appeal of using a readable format for the snapshots but even though I can see the content, I still can't actually verify its accuracy without running a tool over it.

So I think I'd just keep it simple and only support SSZ.

Reply:
For context, I don't really see users manually specifying these snapshots - it's just too much hassle for them when it can easily enough be loaded from the EL anyway. It will almost always just be using the default snapshot we provide.